### PR TITLE
Remove configuration options deprecated in 2.2

### PR DESCRIPTION
**Pull request by pracucci**
#### What this PR does
In this PR I'm removing configuration options we deprecated in 2.2 and announced they would have been removed in 2.4.

I will rebase after https://github.com/grafana/mimir/pull/3110 is merged.

#### Which issue(s) this PR fixes or relates to

N/A

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

**Review by pstibrany — APPROVED** [review]

**Comment by pstibrany** [comment]
Feel free to close my duplicate PR and keep only this one.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,9 @@
 * [CHANGE] Anonymous usage statistics tracking: added the minimum and maximum value of `-ingester.out-of-order-time-window`. #2940
 * [CHANGE] The default hash ring heartbeat period for distributors, ingesters, rulers and compactors has been increased from `5s` to `15s`. Now the default heartbeat period for all Mimir hash rings is `15s`. #3033
 * [CHANGE] Reduce the default TSDB head compaction concurrency (`-blocks-storage.tsdb.head-compaction-concurrency`) from 5 to 1, in order to reduce CPU spikes. #3093
+* [CHANGE] Limits: removed the deprecated YAML configuration option `active_series_custom_trackers_config`. Please use `active_series_custom_trackers` instead. #3111
+* [CHANGE] Ingester: removed the deprecated configuration option `-ingester.ring.join-after`. #3111
+* [CHANGE] Querier: removed the deprecated configuration option `-querier.shuffle-sharding-ingesters-lookback-period`.The value of `-querier.query-ingesters-within` is now used internally for shuffle sharding lookback, while you can use `-querier.shuffle-sharding-ingesters-enabled` to enable or disable shuffle sharding on the read path. #3111
```
Comment (dimitarvdimitrov) [inline CHANGELOG.md:17]:
```suggestion
* [CHANGE] Querier: removed the deprecated configuration option `-querier.shuffle-sharding-ingesters-lookback-period`. The value of `-querier.query-ingesters-within` is now used internally for shuffle sharding lookback, while you can use `-querier.shuffle-sharding-ingesters-enabled` to enable or disable shuffle sharding on the read path. #3111
```

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM, thanks for making the change to mimirtool too

File: CHANGELOG.md [hunk 2]
```diff
@@ -12,8 +12,10 @@
 * [CHANGE] Anonymous usage statistics tracking: added the minimum and maximum value of `-ingester.out-of-order-time-window`. #2940
 * [CHANGE] The default hash ring heartbeat period for distributors, ingesters, rulers and compactors has been increased from `5s` to `15s`. Now the default heartbeat period for all Mimir hash rings is `15s`. #3033
 * [CHANGE] Reduce the default TSDB head compaction concurrency (`-blocks-storage.tsdb.head-compaction-concurrency`) from 5 to 1, in order to reduce CPU spikes. #3093
-* [CHANGE] Previously deprecated `active_series_custom_trackers_config` has been removed. #3110
```
Comment (pracucci) [inline CHANGELOG.md:15]: Rephrased below.

**Pull request merged**